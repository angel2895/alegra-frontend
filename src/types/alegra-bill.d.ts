export interface AlegraBill {
  date: String;
  dueDate: String;
  client: number;
  operationType: string;
  seller: SellerBill;
  items: Item[];
}

export interface Item {
  id: number;
  price: number;
  quantity: number;
}

export interface SellerBill {
  id: String;
}
