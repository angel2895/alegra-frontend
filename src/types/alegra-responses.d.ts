export interface AlegraResponseProduct {
  id: string;
  category: Category;
  name: string;
  description: string;
  reference: string;
  status: string;
  price: Price[];
  inventory: Inventory;
  tax: any[];
  customFields: any[];
  productKey: null;
  type: string;
  itemType: null;
  settingsOnPos: SettingsOnPos;
}

export interface Category {
  id: string;
  name: string;
}

export interface Inventory {
  unit: string;
}

export interface Price {
  idPriceList: number;
  name: string;
  type: string;
  price: string;
  currency: Currency;
}

export interface Currency {
  code: string;
  symbol: string;
}

export interface SettingsOnPos {
  hide: null;
}

export interface AlegraResponseBill {
  id: string;
  date: Date;
  dueDate: Date;
  datetime: Date;
  observations: null;
  anotation: null;
  termsConditions: string;
  status: string;
  client: Client;
  numberTemplate: NumberTemplate;
  subtotal: number;
  discount: number;
  tax: number;
  total: number;
  totalPaid: number;
  balance: number;
  decimalPrecision: string;
  warehouse: Warehouse;
  operationType: string;
  seller: Seller;
  priceList: PriceList;
  items: Item[];
  costCenter: null;
}

export interface Client {
  id: string;
  name: string;
  identification: string;
  phonePrimary: null;
  phoneSecondary: null;
  fax: null;
  mobile: string;
  email: string;
  address: Address;
}

export interface Address {
  state: string;
  city: string;
  district: string;
  urbanization: null;
  ubigeoCode: string;
  address: string;
}

export interface Item {
  name: string;
  description: null;
  price: number;
  discount: number;
  reference: null;
  quantity: number;
  id: number;
  productKey: null;
  unit: Unit;
  tax: any[];
  total: number;
}

export enum Unit {
  Unit = "unit",
}

export interface NumberTemplate {
  id: string;
  prefix: string;
  number: string;
  text: null;
  documentType: string;
  fullNumber: string;
  formattedNumber: string;
}

export interface PriceList {
  id: number;
  name: string;
}

export interface Seller {
  id: number;
  name: string;
  identification: string;
  observations: string;
}

export interface Warehouse {
  id: string;
  name: string;
}
