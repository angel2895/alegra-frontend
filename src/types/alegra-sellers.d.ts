export interface AlegraSeller {
  id: number;
  name: string;
  identification: string;
  observations: string;
  status: Status;
}

export enum Status {
  Active = "active",
}
