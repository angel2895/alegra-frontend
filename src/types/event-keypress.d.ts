export interface EventKeyPress {
  isTrusted: boolean;
  altKey: boolean;
  bubbles: boolean;
  cancelBubble: boolean;
  cancelable: boolean;
  charCode: Number;
  code: String;
}
