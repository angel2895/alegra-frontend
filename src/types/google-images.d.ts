export interface GoogleImages {
  search_metadata: SearchMetadata;
  search_parameters: SearchParameters;
  search_information: SearchInformation;
  suggested_searches: SuggestedSearch[];
  images_results: GoogleImage[];
}

export interface GoogleImage {
  position: number;
  thumbnail: string;
  source: string;
  title: string;
  link: string;
  original: string;
  is_product: boolean;
  in_stock?: boolean;
}

export interface SearchInformation {
  image_results_state: string;
  query_displayed: string;
}

export interface SearchMetadata {
  id: string;
  status: string;
  json_endpoint: string;
  created_at: string;
  processed_at: string;
  google_url: string;
  raw_html_file: string;
  total_time_taken: number;
}

export interface SearchParameters {
  engine: string;
  q: string;
  google_domain: string;
  ijn: string;
  device: string;
  tbm: string;
}

export interface SuggestedSearch {
  name: string;
  link: string;
  chips: string;
  serpapi_link: string;
  thumbnail: string;
}
