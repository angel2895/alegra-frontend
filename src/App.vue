<template>
  <header-component></header-component>
  <main>
    <router-view />
  </main>
  <footer-component></footer-component>
</template>

<script lang="ts">
// Libraries
import { Options, Vue } from "vue-class-component";
import { mapActions } from "vuex";
import { mapGetters } from "vuex";
// Types
import { GoogleImage } from "@/types/google-images";
import { AlegraSeller } from "@/types/alegra-seller";
import { SellRegisterShort } from "@/types/sell-register";

// Components
import HeaderComponent from "@/components/shared/HeaderComponent.vue";
import FooterComponent from "@/components/shared/FooterComponent.vue";

@Options({
  watch: {
    getScores(currentScore: SellRegisterShort[]) {
      console.log("Scores -> ", currentScore);

      let areThereWinner = currentScore.some((sellShort) => {
        return sellShort.points > this.getObjetivePoints;
      });

      if (areThereWinner) {
        let winner = currentScore.find((sellShort) => {
          return sellShort.points > this.getObjetivePoints;
        });
        console.log("Winner -> ", winner);
      }
    },
  },
  components: {
    HeaderComponent,
    FooterComponent,
  },
  methods: mapActions(["getImagesGoogle", "getAlegraSellers"]),
  computed: mapGetters(["getScores", "getObjetivePoints"]),
})
export default class App extends Vue {
  getImagesGoogle!: () => Promise<GoogleImage[]>;
  getAlegraSellers!: () => Promise<AlegraSeller[]>;

  mounted(): void {
    this.getImagesGoogle();
    this.getAlegraSellers();
    window.localStorage.clear();
  }
}
</script>
