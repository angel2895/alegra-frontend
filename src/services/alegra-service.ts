import axios from "axios";

import { AlegraSeller } from "@/types/alegra-seller";

export const urlBaseAlegra = "https://api.alegra.com/api/v1";
export const username = "luisangel2895@gmail.com";
export const password = "c7bc72f6a504039ba712";

// GET Sellers
export const getSellersAlegra = async (): Promise<AlegraSeller[]> => {
  const { data } = await axios.get(`${urlBaseAlegra}/sellers`, {
    auth: {
      username: `${username}`,
      password: `${password}`,
    },
  });
  return data;
};

// POST Bill
export const postBillAlegra = async (): Promise<any> => {
  const response = await axios.post(
    `${urlBaseAlegra}/invoices`,
    {
      date: "2015-11-15",
      dueDate: "2015-12-15",
      client: 2,
      items: [
        {
          id: 1,
          price: 120,
          quantity: 5,
        },
        {
          id: 2,
          description: "Cartera de cuero color caf√©",
          price: 85,
          discount: 10,
          tax: [
            {
              id: 6,
            },
          ],
          quantity: 1,
        },
      ],
    },
    {
      auth: {
        username: `${username}`,
        password: `${password}`,
      },
    }
  );
  return response;
};
