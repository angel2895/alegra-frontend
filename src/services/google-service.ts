import axios from "axios";

import { GoogleImages } from "@/types/google-images";

// Google API = "https://serpapi.com/search.json";

export const urlProxyGoogle = "/api_google";
export const defaultWord = "apple";

export const getGoogleImages = async (
  word: string = defaultWord
): Promise<GoogleImages> => {
  const { data } = await axios.get(`${urlProxyGoogle}/${word}`);
  return data;
};
