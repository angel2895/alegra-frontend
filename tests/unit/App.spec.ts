import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
// Components
import App from "@/App.vue";
// Functions
import { getGoogleImages } from "@/services/google-service";
import { getSellersAlegra } from "@/services/alegra-service";
// Types
import { SellRegister, SellRegisterShort } from "@/types/sell-register";
import { GoogleImage } from "@/types/google-images";
import { AlegraSeller } from "@/types/alegra-seller";

const store = createStore({
  state() {
    return {
      sellRegisters: [] as SellRegister[],
      points_objective: 20 as number,
    };
  },
  getters: {
    getScores: (state) => {
      return (state as any).sellRegisters.map((register: SellRegister) => {
        const tuple: SellRegisterShort = {
          id: 0,
          name: "",
          points: 0,
        };
        tuple.id = register.id;
        tuple.name = register.name;
        tuple.points = register.points;
        return tuple;
      });
    },
    getObjetivePoints: (state) => (state as any).points_objective,
  },
  mutations: {
    UPDATE_GOOGLE_IMAGES(state, payload: GoogleImage[]) {
      (state as any).images = payload;
    },

    UPDATE_LOADING_STATUS(state, payload: boolean) {
      (state as any).isLoading = payload;
    },

    GET_ALEGRA_SELLERS(state, payload: AlegraSeller[]) {
      (state as any).sellers = payload;
    },
  },
  actions: {
    async getImagesGoogle({ commit }, payload: string) {
      commit("UPDATE_LOADING_STATUS", true);
      const googleImages = await getGoogleImages(payload);
      const { images_results } = googleImages;
      commit("UPDATE_GOOGLE_IMAGES", images_results as GoogleImage[]);
      commit("UPDATE_LOADING_STATUS", false);
    },
    async getAlegraSellers({ commit }) {
      const alegraSellers = await getSellersAlegra();
      commit("GET_ALEGRA_SELLERS", alegraSellers);
    },
  },
});

describe("App.vue", () => {
  it("Match with snapshot", () => {
    const wrapper = shallowMount(App, {
      global: {
        plugins: [store],
        stubs: ["router-link", "router-view"],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("HeaderComponent and FooterComponent are mounted in App", () => {
    const wrapper = shallowMount(App, {
      global: {
        plugins: [store],
        stubs: ["router-link", "router-view"],
      },
    });
    expect(wrapper.find("header-component-stub").exists()).toBeTruthy();
    expect(wrapper.find("footer-component-stub").exists()).toBeTruthy();
  });
  it("Called action (getImagesGoogle) in mounted", () => {
    const getImagesGoogleSpy = jest.spyOn<any, any>(
      App.methods,
      "getImagesGoogle"
    );

    shallowMount(App, {
      global: {
        plugins: [store],
        stubs: ["router-link", "router-view"],
      },
    });
    expect(getImagesGoogleSpy).toHaveBeenCalled();
  });
  it("Called action (getAlegraSellers) in mounted", () => {
    const getAlegraSellersSpy = jest.spyOn<any, any>(
      App.methods,
      "getAlegraSellers"
    );

    shallowMount(App, {
      global: {
        plugins: [store],
        stubs: ["router-link", "router-view"],
      },
    });
    expect(getAlegraSellersSpy).toHaveBeenCalled();
  });
});
