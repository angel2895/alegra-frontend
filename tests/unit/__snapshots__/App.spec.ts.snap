// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Match with snapshot 1`] = `
<header-component-stub></header-component-stub>
<main>
  <router-view-stub></router-view-stub>
</main>
<footer-component-stub></footer-component-stub>
`;
